app-typescript/components/DatePicker.tsx

```diff
@@ -334,3 +334,41 @@ export class DatePickerISO extends React.PureComponent<IDatePickerISO> {
         );
     }
 }
+
+const MonthNavigator = ({
+    value,
+    options,
+    onChange,
+}: React.ComponentProps<NonNullable<CalendarProps['monthNavigatorComponent']>>) => (
+    <TreeSelect<string>
+        kind="synchronous"
+        value={[value]}
+        getOptions={() => options.map((o) => ({...o, value: o.id}))}
+        getLabel={(option) => options[parseInt(option)].label}
+        getId={(option) => option}
+        onChange={(selected) => {
+            onChange(selected[0]);
+        }}
+        clearable={false}
+        labelHidden
+    />
+);
+
+const YearNavigator = ({
+    value,
+    options,
+    onChange,
+}: React.ComponentProps<NonNullable<CalendarProps['yearNavigatorComponent']>>) => (
+    <TreeSelect<string>
+        kind="synchronous"
+        value={[value]}
+        getOptions={() => options.map((o) => ({...o, value: o.label}))}
```

Comment:
Do not use naming shorhands - use `option` instead of `o`.

Another issue is that your return value doesn't match the interface. Due to use of spread operator typescript isn't flagging it. That's a good reason to avoid spreads.